runtime/doc/lua.txt

```diff
@@ -714,6 +714,18 @@ vim.mpack.encode({obj})                                     *vim.mpack.encode*
 vim.mpack.decode({str})                                     *vim.mpack.decode*
     Decodes (or "unpacks") the msgpack-encoded {str} to a Lua object.
 
+------------------------------------------------------------------------------
+VIM.JSON                                                            *lua-json*
+
+The *vim.json* module provides encoding and decoding of Lua objects to and
+from JSON-encoded strings. Supports |vim.NIL| and |vim.empty_dict()|.
+
+vim.json.encode({obj})                                       *vim.json.encode*
+    Encodes (or "packs") Lua object {obj} as JSON in a Lua string.
+
+vim.json.decode({str})                                       *vim.json.decode*
```

Comment:
Decode also takes a second options parameter with `luanil = { object: bool, array: bool }` to control if null values should use vim.NIL or nil.

See https://github.com/neovim/neovim/pull/15854/files